dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/hibernate/HibernateProgramInstanceStore.java

```diff
@@ -328,6 +330,39 @@ public void hardDelete( ProgramInstance programInstance )
         getSession().delete( programInstance );
     }
 
+    @Override
+    public List<ProgramInstance> getByProgramAndTrackedEntityInstance(
+        List<Pair<Program, TrackedEntityInstance>> programTeiPair, ProgramStatus programStatus )
+    {
+        if ( programTeiPair == null || programTeiPair.isEmpty() )
```

Comment:
Does it make sense to send in null here? Might it be better to not allow null, since null might indicate a bug? Use checkNotNull and cast a np?

Reply:
Good point, fixed